pkg/secrets/secrets.go

```diff
@@ -17,7 +33,52 @@ func ValidateCredentials(secret *v1.Secret) error {
 		return ValidateAWSCredentials(secret)
 	case AzureSecretType:
 		return ValidateAzureCredentials(secret)
+	case GCPSecretType:
+		return ValidateGCPCredentials(secret)
 	default:
```

Comment:
We may not have a credentials secret for filesystem types but let's confirm validation does not fail for the same.